ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -1541,13 +1541,29 @@ FEDataManager::interpWeighted(const int f_data_idx,
     else
     {
         // Extract local form vectors.
-        auto X_petsc_vec = static_cast<PetscVector<double>*>(&X_vec);
+        auto X_petsc_vec = dynamic_cast<PetscVector<double>*>(&X_vec);
```

Comment:
We should probably rewrite all of this to just take in `PetscVector`s instead of `NumericVector`s.

Reply:
We might as well since we already `static_cast` to that derived class: no one's code is going to work if they supply any other vector here.